platforms/core-runtime/functional/src/main/java/org/gradle/internal/Combiners.java

```diff
@@ -24,9 +24,7 @@
  * Promote to a more shared subproject if useful.
  */
 public abstract class Combiners {
-    private static final BinaryOperator<?> NON_COMBINING = (a, b) -> {
-        throw new IllegalStateException("Not a combinable operation");
-    };
+    private static final BinaryOperator<?> NON_COMBINING = Combiners::apply;
```

Comment:
🤔 why is this lambda cached, but others don't?

Reply:
I guess that would be for @lptr to answer, I'm just avoiding errorprone complaints here :)

Reply:
I'd just inline this.